src/Examples/Sequence/DerivedFormsRBT.agda

```diff
@@ -90,53 +125,266 @@ module MapReduce {A B : tp⁺} where
                   bind cost (g s₁ s₃) λ _ →
                     ret triv
         )
-      ≤⟨ (
-        {!   !}
-        -- ≤⁻-mono {Π _ λ _ → cost} {cost} (bind cost (mapreduce _ _ _ t₁ f g z ∥ mapreduce _ _ _ t₂ f g z))
-        --   {λ (s₁ , s₂) →
-        --     bind cost (f a) λ b →
-        --       bind cost (g b s₂) λ s₃ →
-        --        funext
-        --   $ λ-mono-≤⁻ λ (s₁ , s₂) →
-        -- ≤⁻-mono (bind cost (f a)) $ λ-mono-≤⁻ λ b →
-        -- ≤⁻-mono (bind cost (g b s₂)) $ λ-mono-≤⁻ λ s₃ →
-        -- g-bound {s₁} {s₃}
-      ) ⟩
+      ≤⟨ bind-monoʳ-≤⁻ ((mapreduce/seq _ _ _ t₁ f g z) ∥ (mapreduce/seq _ _ _ t₂ f g z)) (λ (s₁ , s₂) →
+           bind-monoʳ-≤⁻ (f a) (λ b →
+             bind-monoʳ-≤⁻ (g b s₂) (λ s₃ → g-bound))) ⟩
```

Comment:
The problem we ran into earlier can (and should) be solved via `bind-monoʳ-≤⁻` and `bind-monoˡ-≤⁻`.